src/tarantool-gdb.py

```diff
@@ -0,0 +1,468 @@
+            raise gdb.GdbError('list is missing')
+
+        # if entry type is missing try to identify it from the list expression
+        if entry_info is None:
+            def split_member(expr):
+                arrow_pos = expr.rfind('->')
+                dot_pos = expr.rfind('.')
+                member_sep = '->' if arrow_pos > dot_pos else '.'
+                member_container_expr, _, member = expr.rpartition(member_sep)
+                return (member_container_expr, member)
+            member_container_expr, member = split_member(argv[argi])
+            log('member_container_expr={} member={}\n'.format(member_container_expr, member))
+            if member_container_expr:
+                member_container = gdb.parse_and_eval(member_container_expr)
+                if member_container.type.code == gdb.TYPE_CODE_PTR:
+                    member_container = member_container.dereference()
+                log('member_container: {}\n'.format(str(member_container)))
+                entry_info = list_containers_map[member_container.type.tag][member]
+            else:
+                entry_info = list_variables_map[argv[argi]]
+            log('entry_info: {}\n'.format(str(entry_info)))
+
+        if entry_info is None:
+            raise gdb.GdbError('failed to identify entry type')
+
+        list_printer = TtListPrinter(gdb.parse_and_eval(argv[argi]), entry_info)
+        argi += 1
+
+        try:
+            rlist_printer_gen.enabled = False
+
+            if argi < argc:
+                if argv[argi] == 'where':
+                    if argi + 3 >= argc:
+                        raise gdb.GdbError('invalid selector: not enough arguments')
+                    field = argv[argi + 1]
+                    op = argv[argi + 2]
+                    field_value = argv[argi + 3]
+                    log('field={} op={} val={}\n'.format(field, op, field_value));
+                    for index, item in list_printer.children():
+                        log('checking entry: {}\n'.format(str(entry)))
+                        entry = list_printer.list.entry_info.entry_from_item(item)
+                        if operator.attrgetter(op)(operator)(entry[field].string(), field_value):
+                            gdb.write(list_item_str(item, index, list_printer.list.entry_info))
+
+                else:
+                    argc = len(argv)
+                    log('argc={}\n'.format(argc))
+                    while argi < argc:
+                        log('argi={}\n'.format(argi))
+                        index, item = list_printer[int(argv[argi])]
+                        gdb.write(list_item_str(item, index, list_printer.list.entry_info))
+                        argi += 1
+
+            else:
+                gdb.write(list_printer.to_string() + '\n')
+                for index, item in list_printer.children():
+                    gdb.write(list_item_str(item, index, list_printer.list.entry_info))
+
+        finally:
```

Comment:
try-finally approach is used here to execute code that re-enable rlist pretty-printer regardless of the result of the try-block (the mentioned pretty-printer is disabled right before the try-block to avoid recursive printing of a list inside the try-block)
what approach could be used instead?

Reply:
Finally this part has been changed completely and the problem gone